libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -0,0 +1,507 @@
+    }
+
+    if( returnStatus == TRANSPORT_SOCKET_STATUS_SUCCESS )
+    {
+        /* Create a new TCP socket. */
+        tcpSocket = SOCKETS_Socket( SOCKETS_AF_INET,
+                                    SOCKETS_SOCK_STREAM,
+                                    SOCKETS_IPPROTO_TCP );
+
+        if( tcpSocket == ( Socket_t ) SOCKETS_INVALID_SOCKET )
+        {
+            LogError( ( "Failed to create new socket.\n" ) );
+            returnStatus = TRANSPORT_SOCKET_STATUS_INSUFFICIENT_MEMORY;
+        }
+    }
+
+    if( returnStatus == TRANSPORT_SOCKET_STATUS_SUCCESS )
+    {
+        /* Set up connection encryption if credentials are provided. */
+        if( pSocketsConfig->enableTls )
+        {
+            if( ( int32_t ) SOCKETS_ERROR_NONE != tlsSetup( pSocketsConfig,
+                                                            tcpSocket,
+                                                            ( uint8_t * ) pServerInfo->pHostName,
+                                                            hostnameLength ) )
+            {
+                returnStatus = TRANSPORT_SOCKET_STATUS_INVALID_CREDENTIALS;
+            }
+        }
+    }
+
+    if( returnStatus == TRANSPORT_SOCKET_STATUS_SUCCESS )
+    {
+        /* Establish connection. */
+        serverAddress.ucSocketDomain = SOCKETS_AF_INET;
+        serverAddress.usPort = SOCKETS_htons( pServerInfo->port );
+        serverAddress.ulAddress = SOCKETS_GetHostByName( pServerInfo->pHostName );
+
+        /* Check for errors from DNS lookup. */
+        if( serverAddress.ulAddress == ( uint32_t ) 0 )
+        {
+            LogError( ( "Failed to resolve %s.", pServerInfo->pHostName ) );
+            returnStatus = TRANSPORT_SOCKET_STATUS_DNS_FAILURE;
+        }
+    }
+
+    if( returnStatus == TRANSPORT_SOCKET_STATUS_SUCCESS )
+    {
+        TransportSocketStatus = SOCKETS_Connect( tcpSocket,
+                                                 &serverAddress,
+                                                 sizeof( SocketsSockaddr_t ) );
+
+        if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to establish new connection. Socket status: %d.", TransportSocketStatus ) );
+            returnStatus = TRANSPORT_SOCKET_STATUS_CONNECT_FAILURE;
+        }
+    }
+
+    if( returnStatus == TRANSPORT_SOCKET_STATUS_SUCCESS )
+    {
+        TransportSocketStatus = transportTimeoutSetup( tcpSocket, pSocketsConfig->sendTimeoutMs, pSocketsConfig->recvTimeoutMs );
+
+        if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Setup transport timeout configuration failed status: %d.", TransportSocketStatus ) );
+            returnStatus = TRANSPORT_SOCKET_STATUS_INTERNAL_ERROR;
+        }
+    }
+
+    if( returnStatus == TRANSPORT_SOCKET_STATUS_SUCCESS )
+    {
+        pNetworkContext->pContext = tcpSocket;
+        ( void ) pNetworkContext;
```

Comment:
Why is this required?

Reply:
removed in b719c48142e222a8b58ca00e4d06027d03c6e5da